index.js

```diff
@@ -17,49 +26,52 @@ const prepDeploy = (names, stages) => {
 }
 
 const filterObject = (names, stages) => {
-	let parent = [];
-	if (!names) {parent = obj}
+	let tempArray = [];
+	let returnArray = []
 	// filter the names
-	names.forEach(x => {
+	!names ? tempArray = obj : names.forEach(x => {
 		const params = obj.filter(param => param.name === x);
 		if(!params.length) {
-			console.warn('No params found by name:', x);
+			console.warn('No names found by:', x);
 			return;
 		};
-		parent.push(params[0])
+		tempArray.push(params[0])
 	});
-	let arr = []
 	if(!stages){
 		// build array of params
-		parent.forEach(y => {
+		tempArray.forEach(y => {
 			for (const key in y) {
 				if(Array.isArray(y[key])) {
-					arr = arr.concat(y[key]);
+					returnArray = returnArray.concat(y[key]);
 				}
 			}
+		}); 
+		return returnArray;
+	} else {
+		//filter the stages
+		stages.forEach(x => {
+			tempArray.forEach(y => {
+				if(y[x] === undefined || !y[x].length) {
+					console.warn('No stages found by:', x)
```

Comment:
Does this work the same?
```suggestion
					console.warn(`No stages found by: ${x}`)
```

Reply:
yeah it does